pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +92,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthPlugin is a custom plugin used to authenticate to the kubernetes cluster.
+	AuthPlugin string `json:"auth-plugin,omitempty"`
```

Comment:
I would kind of expect to be able to have a custom auth plugin that uses username/password that overrides the default basic auth flow.